### PR TITLE
test: only use emulator for low-level HTTP tests

### PR DESCRIPTION
**Pull request by coryan**
We do not gain more information by using httpbin.org, and that fails sometimes.

Part of the work for #10413

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10961)

### CODE REVIEW

File: google/cloud/storage/tests/curl_request_integration_test.cc [hunk 1]
```diff
@@ -354,12 +365,10 @@ TEST(CurlRequestTest, UserAgent) {
 }
 
 #if CURL_AT_LEAST_VERSION(7, 43, 0)
-bool UsingEmulator() {
-  return google::cloud::internal::GetEnv("HTTPBIN_ENDPOINT").has_value();
-}
```
Comment (devbww) [inline google/cloud/storage/tests/curl_request_integration_test.cc:368]: Remove extra blank line.

File: google/cloud/storage/tests/curl_download_request_integration_test.cc [hunk 2]
```diff
@@ -30,16 +30,16 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
...
 using ::testing::Pair;
 using ::testing::StartsWith;
 
-std::string HttpBinEndpoint() {
-  return google::cloud::internal::GetEnv("HTTPBIN_ENDPOINT")
-      .value_or("https://httpbin.org");
-}
+std::string HttpBinEndpoint() { return GetEnv("HTTPBIN_ENDPOINT").value(); }
+
+bool UsingEmulator() { return GetEnv("HTTPBIN_ENDPOINT").has_value(); }
```
Comment (devbww) [inline google/cloud/storage/tests/curl_download_request_integration_test.cc:42]: Opt Nit{sup N}: I might reverse the declaration order of these to somehow match the `has_value()` before `value()` ordering.

**Review by devbww — APPROVED** [review]

File: google/cloud/storage/tests/curl_request_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/tests/curl_request_integration_test.cc]: Done.

File: google/cloud/storage/tests/curl_download_request_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/tests/curl_download_request_integration_test.cc]: Hmmm... Okay, done.

**Pull request merged**